### PR TITLE
impala: init at 0.1.1

### PR DESCRIPTION
**Pull request by nydragon**
## Description of changes

Addition of [impala](https://github.com/pythops/impala).

Impala is a TUI for managing wifi through iwd.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/im/impala/package.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  src = fetchFromGitHub {
+    owner = "pythops";
+    repo = "impala";
+    rev = "v${version}";
+    hash = "sha256-r/aWzSn/Dci69oS/yopG6Ro34U8hniHVanctyM7RvDw=";
+  };
+
+  cargoHash = "sha256-IV1ftsRyM0CUlQMVmLip1FiqnouT5TsKASpF/KLARqY=";
+
+  meta = with lib; {
```
Comment (AndersonTorres) [inline pkgs/by-name/im/impala/package.nix:19]:
```suggestion
  meta = {
```

File: pkgs/by-name/im/impala/package.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    description = "TUI for managing wifi";
+    homepage = "https://github.com/pythops/impala";
+    license = licenses.gpl3Only;
+    maintainers = [ maintainers.nydragon ];
```
Comment (AndersonTorres) [inline pkgs/by-name/im/impala/package.nix:23]:
```suggestion
    license = lib.licenses.gpl3Only;
    maintainers = [ lib.maintainers.nydragon ];
```

**Comment by matteo-pacini** [comment]
The tool needs a better error handling...

```
[nix-shell:~/.cache/nixpkgs-review/pr-319553]$ impala
thread 'main' panicked at /build/impala-0.1.1-vendor.tar.gz/iwdrs/src/session.rs:35:56:
called `Result::unwrap()` on an `Err` value: MethodError(OwnedErrorName("org.freedesktop.DBus.Error.ServiceUnknown"), Some("The name net.connman.iwd was not provided by any .service files"), Msg { type: Error, serial: 3, sender: UniqueName("org.freedesktop.DBus"), reply-serial: 2, body: Signature("s"), fds: [] })
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

File: pkgs/by-name/im/impala/package.nix (hunk 2 above)
Comment (matteo-pacini) [inline pkgs/by-name/im/impala/package.nix:23]: It would be great to restrict platforms to the same  `iwd` supports

**Review by matteo-pacini — CHANGES_REQUESTED** [review]

File: pkgs/by-name/im/impala/package.nix (hunk 2 above)
Comment (Nydragon) [reply pkgs/by-name/im/impala/package.nix]: Addressed the suggestion!

Comment (Nydragon) [reply pkgs/by-name/im/impala/package.nix]: Addressed the suggestion!

**Review by matteo-pacini — APPROVED** [review]
LGTM

**Review by h7x4 — APPROVED** [review]

**Pull request merged**